tokio/src/process/mod.rs

```diff
@@ -1239,6 +1245,81 @@ impl Child {
             stderr,
         })
     }
+
+    /// Starts streaming stdout and stderr into unbounded channels, in addition to a future that will eventually
+    /// resolve to the exit code for the process. If these queues are not received from, they will
+    /// cause memory usage to grow forever. This function is most useful to people who need to stream stdout and
+    /// stderr, and have no tolerance for the child process blocking due to running out of space in the stdio buffers.
+    #[cfg(all(feature = "sync", feature = "io-util", feature = "rt-multi-thread"))]
+    pub fn unbounded_streaming(mut self) -> ChildUnboundedStream {
+        use crate::io::{AsyncReadExt, AsyncWriteExt};
+        use crate::spawn;
+
+        let stdin_pipe = self.stdin.take();
+        let stdout_pipe = self.stdout.take();
+        let stderr_pipe = self.stderr.take();
+        let (stdin_sender, mut stdin_receiver) = unbounded_channel::<Vec<u8>>();
+        let (stdout_sender, stdout_receiver) = unbounded_channel();
+        let (stderr_sender, stderr_receiver) = unbounded_channel();
+        let (exit_sender, exit_receiver) = oneshot::channel();
+
+        async fn unbounded_buffers<P>(mut pipe: P, sender: UnboundedSender<Vec<u8>>)
+        where
+            P: AsyncRead + Unpin,
+        {
+            loop {
+                let mut v = vec![];
+                if pipe.read_buf(&mut v).await.is_ok() {
+                    if sender.send(v).is_err() {
+                        break;
+                    }
+                } else {
+                    break;
+                }
+            }
+        }
```

Comment:
I think there's an opportunity for a race condition here. Just because `Child::wait()` has finished doesn't mean all of the pipe has been read. Ideally once the program closes these futures would do a `read_to_end` on their pipe, and then close the channel by dropping the sender.

However I'm not sure how to implement this without using something like `tokio::select!`, and  that macro isn't available in this context. Plus it would be yet another feature added to the already large list of features required for this to work.

Reply:
Really what I need is a `Select` future, though I'm not sure if some better solution already exists in this crate.

Reply:
Nevermind, this supposed problem was created because I believed that stdout/stderr didn't EOF properly when the process terminated. I've seen now that they do, and updated the code to handle this correctly.